qa/os/src/test/java/org/elasticsearch/packaging/test/ArchiveTestCase.java

```diff
@@ -263,6 +263,64 @@ public void test60AutoCreateKeystore() throws Exception {
         });
     }
 
+    public void test61PasswordProtectedKeystore() throws Exception {
+        assumeThat(installation, is(notNullValue()));
+        // cleanup from previous test
+        // rm(installation.config("elasticsearch.keystore"));
+
+        String password = "keystorepass";
+        final Shell sh = new Shell();
+        Platforms.onLinux(() -> {
+            String systemJavaHome = sh.run("echo $SYSTEM_JAVA_HOME").stdout.trim();
+            sh.getEnv().put("JAVA_HOME", systemJavaHome);
+        });
+        Platforms.onWindows(() -> {
+            final String systemJavaHome = sh.run("$Env:SYSTEM_JAVA_HOME").stdout.trim();
+            sh.getEnv().put("JAVA_HOME", systemJavaHome);
+        });
+
+
+        final Installation.Executables bin = installation.executables();
+
+        // set the password by passing it to stdin twice
+        Platforms.onLinux(() ->
+            sh.run("echo $\'" + password + "\n" + password + "\n\' | sudo -u " + ARCHIVE_OWNER + " "
```

Comment:
We seem to only be using stdin, but shouldn't we also be testing input from the console? Probably using `expect`?

Reply:
I've added some tests that use `expect` in an `sh.run()` block. Let me know if a different approach to invoking `expect` would be better.